### ISSUE TITLE
Node.text() does not respect changes from Node.unwrap_tags

### ISSUE THREAD

**Issue by NixBiks** [rushter/selectolax#68]
It seems that `node.text()` does not respect the mutated node after calling `node.unwrap_tags`. I'd expect the following to pass

```python
from selectolax.parser import HTMLParser


tree = HTMLParser("<div><p><strong>J</strong>ohn</p><p>Doe</p></div>")
tree.unwrap_tags(["strong"])
node = tree.css_first("div", strict=True)
assert node.html == "<div><p>John</p><p>Doe</p></div>"
text = tree.text(deep=True, separator=" ", strip=True)
assert text == "John Doe", f"{text} != John Doe"
```

but instead I get

```python
AssertionError: J ohn Doe  != John Doe
```

**Comment by mr-bjerre** [comment]
I have quite the ugly workaround currently (parsing the html twice)

```python
pre_tree = tree = HTMLParser("<div><p><strong>J</strong>ohn</p><p>Doe</p></div>")
pre_tree.unwrap_tags(["strong"])

tree = HTMLParser(pre_tree.html)
node = tree.css_first("div", strict=True)
assert node.html == "<div><p>John</p><p>Doe</p></div>"
text = tree.text(deep=True, separator=" ", strip=True)
assert text == "John Doe", f"{text} != John Doe"

**Comment by rushter** [comment]
That happens because you get two text nodes close to each other when removing the `strong` tag.
```
for node in tree.root.traverse(include_text=True):
    print(node.text(), node.tag)
JohnDoe html
 head
JohnDoe body
JohnDoe div
John p
J -text
ohn -text
Doe p
Doe -text
```

I'm not sure what to do about it yet. Technically, I think this behaviour is correct, but for the majority of users it would be unexpected.

**Comment by rushter** [comment]
I get the same behavior in Chrome:
![image](https://user-images.githubusercontent.com/3187951/188793014-c7a96839-0a79-46ea-9526-e8314efe5561.png)

![image](https://user-images.githubusercontent.com/3187951/188792947-12575c50-3ff4-44d7-a6d4-2e5d55a3bca8.png)

**Comment by mr-bjerre** [comment]
Ahh I see. I guess I'd need to iterate through the node and merge all `-text` siblings before. Not sure if this should be part of `node.text`?

**Comment by rushter** [comment]
I think adding a separate ``merge_text_nodes` method will be sufficient.

**Comment by rushter** [comment]
I've made a new release that supports ``merge_text_nodes``.

**Comment by mr-bjerre** [comment]
@rushter you're a legend !

**Comment by mr-bjerre** [comment]
Shouldn't it be added to `parser.pyi` as well?

**Comment by rushter** [comment]
Yes, I've updated it. Thank you for checking it.

**Comment by rushter** [comment]
> I'm afraid there is a memory issue with this thing:
> 
> > python(94018,0x101508580) malloc: Incorrect checksum for freed object 0x129e29870: probably modified after being freed.
> > Corrupt value: 0xb0000000129e3f00
> > python(94018,0x101508580) malloc: *** set a breakpoint in malloc_error_break to debug
> 
> This is what I apply to some html
> 
> ```python
> def html_to_text(html: str):
>     tree = HTMLParser(html)
>     tree.unwrap_tags(DEFAULT_UNWRAP_TAGS)
>     tree.strip_tags(DEFAULT_REMOVE_TAGS)
>     body = tree.body
>     if body is None:
>         raise Exception("No body")
>     body.merge_text_nodes()
>     return body.text(separator="\n", strip=True)
> ```
> 
> I'm running on Macbook Pro M1 - not sure if that is relevant


Is it possible to provide the HTML that crashes it?

**Comment by mr-bjerre** [comment]
> Is it possible to provide the HTML that crashes it?

Yes but it is inconstent. I can send you a JSON file with 300'ish HTML. When I loop through then it crashes at random times.

**Comment by rushter** [comment]
Ok, please send HTML to me.

**Comment by mr-bjerre** [comment]
> I think I fixed it. Can you please double check by clonning the git repo and installing dev version?

I agree. It works !

**Comment by mr-bjerre** [comment]
I see you released 0.3.11 but is it correct that `merge_text_nodes` is missing in the typings file?
